eth/core/precompile/factories.go

```diff
@@ -150,3 +154,109 @@ func (sf *StatefulFactory) buildIdsToMethods(
 	}
 	return idsToMethods, nil
 }
+
+// GeneratePrecompileMethods generates the methods for the given Precompile's ABI.
+func GeneratePrecompileMethods(abiMethods map[string]abi.Method, contractImpl reflect.Value) (Methods, error) {
+	return suitableMethods(abiMethods, contractImpl)
+}
+
+// This function matches each Go implementation of the Precompile
+// to the ABI's respective function.
+// It first searches for the ABI function in the Go implementation. If no find, then panic.
+// It then performs some basic validation on the implemented function
+// Then, the implemented function's arguments are checked against the ABI's arguments' types.
+func suitableMethods(pcABI map[string]abi.Method, contractImpl reflect.Value) (Methods, error) {
+	contractImplType := contractImpl.Type()
+	var methods Methods
+	for m := 0; m < contractImplType.NumMethod(); m++ { // iterate through all of the impl's methods
+		implMethod := contractImplType.Method(m) // grab the Impl's current method
+
+		if isBaseContractMethodOrUnexported(implMethod) {
+			continue // ignore BaseContract's methods, they clearly won't be in the abi
+			// also ignore methods that are not exported
+		}
+
+		implMethodName := formatName(implMethod.Name)         // make the first letter lowercase
+		if abiMethod, found := pcABI[implMethodName]; found { // if the method is found in the ABI
+			if err := basicValidation(implMethod, abiMethod); err != nil {
+				return nil, err
+			}
+			toExecute := newExecute(implMethod) // grab the actual function
+			methods = append(methods, &Method{
+				AbiMethod: &abiMethod,
+				AbiSig:    abiMethod.Sig,
+				Execute:   toExecute,
+			}) // add it to the list of methods
+		} else {
+			return nil, errors.Wrap(ErrNoPrecompileMethodForABIMethod, implMethodName)
+		}
```

Comment:
just check at the end that the size of `methods` is equal to size of abi methods